### ISSUE TITLE
FIFO priority test case fails sometimes

### ISSUE THREAD

**Issue by dmbarreiro** [agenda/agenda#853]
Sometimes job concurrency test 'should run jobs as first in first out (FIFO) with respect to priority' fails. I think I tracked down the problem to `lib/utils/process-jobs.js`. In function `jobProcessing` when we are about to process the job we use setTimeout to delay the execution till the moment the job was scheduled. In the above mentioned test case we schedule two jobs on the same datetime but with different priorities (one low, the other one high). Jobs are processed correctly according to the priority (first high then low) but when we use setTimeout sometimes the low priority job timeouts first ([this](https://stackoverflow.com/questions/48614436/node-js-are-settimeout-callbacks-fired-in-the-same-order-as-registered) investigation on stackoverflow I get to the conclussion node does not guarantee callback execution order for our case in this test) and gets executed before high priority job.

I've seen some PRs getting blocked by this problem even though it has nothing to do with the PR itself so I think we should either change the test case and readme clearly stating we do not guarantee priority on this scenario or we take a different approach to priority. I submitted PR #852 with one possible solution (commit 9f1e4b5), please check it up.
